core/lib/zksync_core/src/eth_watch/event_processors/set_chain_id.rs

```diff
@@ -0,0 +1,98 @@
+use zksync_dal::StorageProcessor;
+use zksync_types::{web3::types::Log, Address, ProtocolUpgrade, H256};
+
+use crate::eth_watch::{
+    client::{Error, EthClient},
+    event_processors::EventProcessor,
+    metrics::{PollStage, METRICS},
+};
+
+/// Responsible for saving new protocol upgrade proposals to the database.
+#[derive(Debug)]
+pub struct SetChainIDEventProcessor {
+    diamond_proxy_address: Address,
+    set_chain_id_signature: H256,
+}
+
+impl SetChainIDEventProcessor {
+    pub fn new(diamond_proxy_address: Address) -> Self {
+        Self {
+            diamond_proxy_address,
+            set_chain_id_signature: H256::default(),
```

Comment:
we should know the signature already, right?

Reply:
Yes but I didn't want to hardcode it, since it already changed twice only in the last 2 weeks